subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/results/StateTrackingTestResultProcessor.java

```diff
@@ -47,8 +54,26 @@ public final void started(TestDescriptorInternal test, TestStartEvent event) {
         delegate.started(state.test, event);
     }
 
+    private void ensureChildrenCompleted(Object testId, long endTime) {
+        List<Object> incompleteChildren = new LinkedList<Object>();
```

Comment:
There's pretty much no reason to ever use LinkedList. The textbook reason of "it's faster if you only insert/iterate" doesn't apply on modern hardware. LinkedList has bad cache locality and will loose out to an array list on almost every account.

Not that this matters here, but I think it's good to use array list out of reflex :)

Reply:
done